partner_manual_rank/models/res_partner.py

```diff
@@ -72,3 +72,12 @@ def _default_is_customer(self):
 
     def _default_is_supplier(self):
         return self.env.context.get("res_partner_search_mode") == "supplier"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if "customer_rank" in vals and "is_customer" not in vals:
+                vals["is_customer"] = bool(vals["customer_rank"])
+            if "supplier_rank" in vals and "is_supplier" not in vals:
+                vals["is_supplier"] = bool(vals["supplier_rank"])
+        return super().create(vals_list)
```

Comment:
Since it is too much tricky, I propose to entirely delete the `_default_*` callbacks I previously wrote and copy/paste the odoo logic from `odoo/addons/account/models/partner.py`.
That way default value is set at only one place and computed functions will be called accordingly.

```
    @api.model_create_multi
    def create(self, vals_list):
        search_partner_mode = self.env.context.get('res_partner_search_mode')
        is_customer = search_partner_mode == 'customer'
        is_supplier = search_partner_mode == 'supplier'
        if search_partner_mode:
            for vals in vals_list:
                if is_customer and 'is_customer' not in vals:
                    vals['is_customer'] = 1
                elif is_supplier and 'is_customer' not in vals:
                    vals['is_supplier'] = 1
        return super().create(vals_list)
```